### PR TITLE
Register defaultSyncRoot with the socket api

### PR DESCRIPTION
**Pull request by TheOneRing**
The socked api won't populate the conext menu for the directory
but in every other regard it will behave like an ownCloud managed dir.

Fixes: #9838

### CODE REVIEW

File: src/gui/socketapi/socketapi.cpp [hunk 1]
```diff
@@ -77,6 +77,11 @@
 
 namespace {
 
+QString unegisterpathMessage()
```
Comment (fmoc) [inline src/gui/socketapi/socketapi.cpp:80]:
```suggestion
const QString UnegisterPathMessage()
```

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/12302/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/12302/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/12302/guiReportUpload/serverlog.log

Screenshots:
- [Syncing_folders_each_having_500_files.png](https://cache.owncloud.com/public/owncloud/client/12302/guiReportUpload/screenshots/Syncing_folders_each_having_500_files.png)

</details>

File: src/gui/socketapi/socketapi.cpp [hunk 2]
```diff
@@ -77,6 +77,11 @@
 
 namespace {
 
+const QString unegisterpathMessageC()
```
Comment (fmoc) [inline src/gui/socketapi/socketapi.cpp:80]:
```suggestion
const QString unregisterpathMessageC()
```

**Review by fmoc — CHANGES_REQUESTED** [review]

File: src/gui/socketapi/socketapi.cpp [hunk 3]
```diff
@@ -405,7 +432,7 @@ void SocketApi::slotUnregisterPath(Folder *folder)
     if (!_registeredFolders.contains(folder))
         return;
 
-    broadcastMessage(buildMessage(QLatin1String("UNREGISTER_PATH"), removeTrailingSlash(folder->path()), QString()), true);
+    broadcastMessage(buildMessage(unegisterpathMessageC(), Utility::stripTrailingSlash(folder->path()), QString()), true);
```
Comment (fmoc) [inline src/gui/socketapi/socketapi.cpp:435]:
```suggestion
    broadcastMessage(buildMessage(unregisterpathMessageC(), Utility::stripTrailingSlash(folder->path()), QString()), true);
```

File: src/gui/socketapi/socketapi.cpp [hunk 4]
```diff
@@ -390,13 +391,39 @@ void SocketApi::slotReadSocket()
...
+}
+
+void SocketApi::unregisterAccount(const AccountPtr &a)
+{
+    if (!_registeredAccounts.contains(a)) {
+        return;
+    }
+
+    if (a->hasDefaultSyncRoot()) {
+        broadcastMessage(buildMessage(unegisterpathMessageC(), Utility::stripTrailingSlash(a->defaultSyncRoot())));
```
Comment (fmoc) [inline src/gui/socketapi/socketapi.cpp:415]:
```suggestion
        broadcastMessage(buildMessage(unregisterpathMessageC(), Utility::stripTrailingSlash(a->defaultSyncRoot())));
```

**Review by fmoc — APPROVED** [review]

**Pull request merged**